### ISSUE TITLE
imv crashes sometimes

### ISSUE THREAD

**Issue by acoolstraw** [labwc/labwc#412]
[imv](https://sr.ht/~exec64/imv/) crashes sometimes, I'm not sure how exactly to reproduce it, but if you A-Tab to another window and back and do some other stuff with the mouse it crashes with this message:
`Floating point exception (core dumped)`

**Comment by Consolatis** [comment]
I think we still leak a stray PRESS event for the modifier (`A` in this case) to the focused application when doing a keybind involving modifiers.
That could be the issue here but I have no idea what has to do with that floating point exception.

**Comment by bi4k8** [comment]
This is at least in part an imv bug; the below patch for imv treats a zero repeat interval as 1, which is wrong but prevents the crash:
```
diff --git a/src/wl_window.c b/src/wl_window.c
index 5efa42f..4b8dd1c 100644
--- a/src/wl_window.c
+++ b/src/wl_window.c
@@ -239,7 +239,7 @@ static void keyboard_repeat(void *data, struct wl_keyboard *keyboard,
   (void)keyboard;
   struct imv_window *window = data;
   window->repeat_delay = delay;
-  window->repeat_interval = 1000 / rate;
+  window->repeat_interval = 1000 / (rate ? rate : 1);
 }
 
 static const struct wl_keyboard_listener keyboard_listener = {
```

On the labwc side, it's concerning that we send a `repeat_info.rate` of 0. This may have something to do with the key repeat problem [mentioned](https://github.com/labwc/labwc/issues/203#issuecomment-1058205955) in #203.

**Comment by bi4k8** [comment]
Apparently we set the key repeat rate to 0 whenever a keybind is activated:
```C
static bool
handle_keybinding(struct server *server, uint32_t modifiers, xkb_keysym_t sym)
{
	struct keybind *keybind;
	struct wlr_keyboard *kb = &server->seat.keyboard_group->keyboard;
	wl_list_for_each_reverse (keybind, &rc.keybinds, link) {
		if (modifiers ^ keybind->modifiers) {
			continue;
		}
		for (size_t i = 0; i < keybind->keysyms_len; i++) {
			if (xkb_keysym_to_lower(sym) == keybind->keysyms[i]) {
				wlr_keyboard_set_repeat_info(kb, 0, 0);
				actions_run(NULL, server, &keybind->actions, 0);
				return true;
			}
		}
	}
	return false;
}
```

This seems wrong. For one thing, it would be nice to have key repeat for some keybinds (e.g. Alt-Tab). For another, we shouldn't be changing the global keyboard configuration here, we should just be ignoring repeats for keybinds (if desired). Sway doesn't do anything like this--it only calls `wlr_keyboard_set_repeat_info` when a keyboard is added or reconfigured.

**Comment by Consolatis** [comment]
AFAIR the reason for the key repeat rate being 0 is
- #176

**Comment by johanmalm** [comment]
Nice work @bi4k8 
I think the repeat thing is one of the last gremlins in the code base and we ought to get rid of it.
My fault for putting it in as a work around.
I suspect it’s hiding another bug.
I vote for reverting the commit and solving why some apps get repeat letters.

**Comment by Consolatis** [comment]
> I vote for reverting the commit and solving why some apps get repeat letters.

completely agree

**Comment by Consolatis** [comment]
Should be fixed by #438
